src/vmq_reg.erl

```diff
@@ -446,92 +454,85 @@ direct_plugin_exports(Mod) when is_atom(Mod) ->
     %% This Function exports a generic Register, Publish, and Subscribe
     %% Fun, that a plugin can use if needed. Currently all functions
     %% block until the cluster is ready.
-    case {vmq_config:get_env(trade_consistency, false),
-          vmq_config:get_env(default_reg_view, vmq_reg_trie)} of
-        {TradeConsistency, DefaultRegView}
-              when is_boolean(TradeConsistency)
-                   and is_atom(DefaultRegView) ->
-            MountPoint = "",
-            ClientId = fun(T) ->
-                               list_to_binary(
-                                 base64:encode_to_string(
-                                   integer_to_binary(
-                                     erlang:phash2(T)
-                                    )
-                                  ))
-                       end,
+    CAPPublish = vmq_config:get_env(allow_publish_during_netsplit, false),
+    CAPSubscribe = vmq_config:get_env(allow_subscribe_during_netsplit, false),
+    CAPUnsubscribe = vmq_config:get_env(allow_unsubscribe_during_netsplit, false),
+    RegView = vmq_config:get_env(default_reg_view, vmq_reg_trie),
+    MountPoint = "",
+    ClientId = fun(T) ->
+                       list_to_binary(
+                         base64:encode_to_string(
+                           integer_to_binary(
+                             erlang:phash2(T)
+                            )
+                          ))
+               end,
+    CallingPid = self(),
+    SubscriberId = {MountPoint, ClientId(CallingPid)},
+    User = {plugin, Mod, CallingPid},
+
+    RegisterFun =
+    fun() ->
+            PluginPid = self(),
+            wait_til_ready(),
+            PluginSessionPid = spawn_link(
+                                 fun() ->
+                                         monitor(process, PluginPid),
+                                         plugin_queue_loop(PluginPid, Mod)
+                                 end),
+            QueueOpts = maps:merge(vmq_queue:default_opts(),
+                                   #{clean_session => true,
+                                     is_plugin => true}),
+            {ok, _, _} = register_subscriber(PluginSessionPid, SubscriberId,
+                                             QueueOpts, ?NR_OF_REG_RETRIES),
+            ok
+    end,
```

Comment:
The RegisterFun seems to be unconditional and not depending on the allow_register_during_netsplit  - while the other functions (publish/subscribe/...) are depending on their respective flags and can return {error, not_ready} depending on those flags. Is this on purpose? I think this was also the old behaviour - but wanted to ask anyway.